### PR TITLE
Project VertexShader Brooke Isaacson

### PR DESCRIPTION
**Pull request by Brooken2**
screenshot 61

### CODE REVIEW

File: definitions.h [hunk 1]
```diff
@@ -50,6 +51,244 @@ struct Vertex
     double w;
 };
 
+/****************************************************
+ * Matrix for transforming vertices. 
+ ****************************************************/
+class Matrix
+{
+  private:
+    double * ptrM;
```
Comment (Napler) [inline definitions.h]: Great way to make any size array

File: definitions.h [hunk 2]
```diff
@@ -50,6 +51,244 @@ struct Vertex
...
+        {
+            delete [] ptrM;
+            ptrM = NULL;
+        }
+    }
+
+    /*************************************************************************
+     * Add rotation to the current matrix.
+     ************************************************************************/
+    void rotation(double rot1)
```
Comment (Napler) [inline definitions.h]: Eventually we will want rotation on all axes

File: definitions.h [hunk 3]
```diff
@@ -243,6 +477,10 @@ class Attributes
         //Changes to make this class more flexable
         double newColor[16];
         void* ptrImg;
+       // Matrix attrMatrix;
+        //Vertex *vert;
+        Vertex *vert;
```
Comment (Napler) [inline definitions.h]: you shouldn't need to store a vertex in your attributes

**Comment by Bluhm-Alexander** [comment]
Awesome code!
very clean. I also tried to implement a dynamically allocated array but it gave me a weird compiling error. I implemented it the same way you did but it seg faults in the constructor so weird.

I'm going to go back and try to reimplement my dynamic array again. right now it's a static 16 member array. It still works because it's all based on size and boundaries but still it bothers me that it isn't dynamically allocated.

**Pull request merged**